Assets/HoloToolkit/Utilities/Editor/SceneLauncherEditor.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using UnityEngine.SceneManagement;
+
+namespace HoloToolkit.Unity
+{
+    [CustomEditor(typeof(SceneLauncher))]
+    public class SceneLauncherEditor : Editor
```

Comment:
It's an interesting idea, but I don't like the way it is turning out. It adds a lot of complexity to have the code for an editor that is duplicating the build settings functionality, it could be surprising that editing a list in the scene launcher also changes your build settings, and also it is complicated to keep the two lists in sync. Here are some alternatives I would consider:
1) Revert to the previous approach where the user specifies scenes using the build settings via the standard Unity API and using SceneList.cs to persist a list of the scenes when builds are created. Optionally, a custom editor for SceneLauncher could have text informing the user that the scenes that are available can be selected via Build Settings.
1) Maintain a completely different list in SceneLauncher. This isn't exactly what HoloToolkit needs, but some projects might want to have a long list of scenes in the build, but only have a subset of those scene launchable from the SceneLauncher menu. For example, a project that uses additive scene loading where some of the scenes can't run by themselves wouldn't want all of them to be launchable. Giving the SceneLauncher its own list without trying to tie it to the build settings might simplify things. Additionally, it would be nice to have a custom editor for SceneLauncher with a utility buttons for "Add All Scenes from Build Settings" so that users who want everything don't have to manually add scenes one at a time. Unfortunately, scenes are added to HoloToolkit all the time and people probably wouldn't think to go into SceneLauncher and click the button to add their new scene. So SceneLauncher would probably fall out of date and only launch some of the scenes that are available.

What do you think, Stephen?